server/models/procedures/stock.sql

```diff
@@ -274,4 +274,188 @@ BEGIN
   VALUES (HUID(UUID()), documentUuid, depotUuid, lotUuid, fluxId, CURRENT_DATE(), stockLotQuantity, inventoryUnitCost, 0, userId, periodId);
 END $$
 
+-- the main procedure that loops through stock_movement , retrieve and caculate cmm data
+/*
+CALL computeStockQuantity(startDate, inventoryUuid, depotUuid);
+
+DESCRIPTION
+Computes the quantity in stock from the startDate until now for a given inventoryUuid.
+
+Originally written by @jeremielodi
+*/
+DROP PROCEDURE IF EXISTS `computeStockQuantity`$$
+CREATE PROCEDURE `computeStockQuantity` (
+  IN _start_date DATE,
```

Comment:
If there will not be an end date, the best will be to simply use date instead of start date